connectors/asana/element-templates/asana-connector.json

```diff
@@ -42,10 +46,15 @@
   ],
   "properties": [
     {
-      "type": "Hidden",
+      "type": "String",
+      "group": "taskDefinitionType",
+      "description": "Task definition type should correspond to connector's runtime. Details in the <a href=\"https://docs.camunda.io/docs/components/connectors/use-connectors/\" target=\"_blank\">documentation</a>",
```

Comment:
I'm a bit concerned that we want to expose this property for every connector template. It doesn't make sense in SaaS to change the template type and may confuse SaaS users. Self-managed users, on the other hand, should be capable of editing the hidden field, given they have applied a custom properties configuration to the connector runtime.

Reply:
This is what, in my understanding, we agreed last time during discussion with @bastiankoerber. But I am happy to discuss this during architecture session today.

---

> It doesn't make sense in SaaS to change the template type and may confuse SaaS users.

I can say that we have misunderstanding in this statement. I did it specifically for SaaS. Say, if you have a standard cluster: Zeebe + Connectors. You also would like to have an instance of Connectors with some super-critical configuration inside your private network.

If you would like to run your BPMN process through SaaS Connector instance - you do nothing - Keep `io.camunda.myconnector:1`. If you would like to run a BPMN process through your local, then you have to change the `taskDefinitionType` as `io.camunda.myconnector:2` in you diagram.

That's what I showed on demo on Friday.

But again, happy to discuss, maybe I missed some important use-case.

---

Regarding the confusing part. It is a bit from the CX part but we requested a feature to make that group collapsable by default. See this one: https://github.com/camunda/camunda-modeler/issues/3807.

Reply:
Got it - thanks for the clarification, that makes sense. Let's discuss further in the architecture session.

Reply:
As per architecture session, will take it offline with @bastiankoerber. For now I removed all element template changes and if needed, I'll make a separate PR.